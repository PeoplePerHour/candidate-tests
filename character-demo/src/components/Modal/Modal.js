import React from 'react';
import './Modal.scss';

class Modal extends React.Component {
        

    render() {

        
    
        if(this.props.isShown) {

            let heShe = this.props.selectedCharacter.gender === 'Male' ? 'He' :(this.props.selectedCharacter.gender === 'Female' ? 'She' : 'It');
            let beOrNot = this.props.selectedCharacter.status === 'Alive' ? 'lives' : (this.props.selectedCharacter.status === 'Dead' ? 'used to live' : 'probably lives') ;

        return (
            <div>
                <div className="modal-overlay" onClick={this.props.onOverlay} />
                  <div className="modal-content">
                     <div><img src={this.props.selectedCharacter.image} alt="Avatar" style={{width: '100%'}} /></div>
                            <b>{this.props.selectedCharacter.name}</b>
                            <b>{heShe} is {this.props.selectedCharacter.species}</b>
                            {/* <b>{heShe} is {this.props.selectedCharacter.gender}</b> */}
                            <b>{this.props.selectedCharacter.status} status</b>
                            <b>{heShe} {beOrNot} in {this.props.selectedCharacter.location.name}</b>
                        </div>
                   </div>
        );
            // }
        }else {
            return null;
        }

    }

}

export default Modal;