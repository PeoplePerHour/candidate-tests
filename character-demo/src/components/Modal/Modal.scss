.modal-overlay {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    height: 100%;
    width: 100%;
    z-index: 1000;
    background-color: rgba(75, 73, 73, 0.2);
}

.modal-content {
    position: fixed;
    top: 14%;
    right: 0;
    bottom: 0;
    width: 450px;
    left: 34%;
    height: 650px;
    z-index: 10000;
    overflow: auto;
    text-align: center;
    padding: 4px;
    cursor: pointer;
}

.modal-content:after {
    vertical-align: middle;
    display: inline-block;
    height: 100%;
    margin-left: -.05em;
    content: '';
}

.modal-dialog {
    position: relative;
    outline: 0;
    width: auto;
    display: inline-block;
    vertical-align: middle;
    box-sizing: border-box;
    max-width: auto;
    cursor: default;
    border-radius: 4px;
}