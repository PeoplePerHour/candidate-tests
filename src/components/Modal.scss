@import '~styles/colors';

body.ReactModal__Body--open {
  overflow: hidden;
}

.Modal__Overlay {
  background-color: rgba($background-color,.1);
  flex-direction: column;
  display: flex;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 100;
}

.ReactModal__Content {
  position: absolute;
  z-index: 100;
  left: 50%;
  margin: 0 auto;
  top: 20%;
  width: 65%;
  max-width: 1000px;
  transform: translate(-50%);
  background: white;
}

.CharacterDetailModal {
    box-shadow: 1px 1px 10px rgba(0,0,0,.05);

    @media only screen and (max-width: 768px) {
      width: 80%;
    }
}
