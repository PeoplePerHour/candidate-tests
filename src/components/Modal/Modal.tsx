import * as React from "react";

interface Iprops{
  children: React.ReactNode
  closeModal():void
  displayModal(b:boolean):void
}

const Modal= (props:Iprops) => {
  const { children,displayModal } = props;
  function closeModal(e:React.MouseEvent) {
    e.stopPropagation();
    props.closeModal();
  }

  const divStyle = {
    display: props.displayModal ? "flex" : "none"
  };

  return (
    <div className="modal" onClick={closeModal} style={divStyle}>
      <div className="modal-content" onClick={e => e.stopPropagation()}>
        <span className="close" onClick={closeModal}>
          &times;
        </span>
        <div>
        <div className="modal-flex">{displayModal ?children:null}</div></div>
      </div>
      <div className='modal-background' ></div>
    </div>
  );
};

export default Modal;
