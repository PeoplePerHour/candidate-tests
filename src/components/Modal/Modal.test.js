
import * as  React from 'react';
import * as ReactDOM from 'react-dom';

import Modal from './Modal'


it('renders Modal without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Modal/>, div);
    expect(div.firstChild).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
})
