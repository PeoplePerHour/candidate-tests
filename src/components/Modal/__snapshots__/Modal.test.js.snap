// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Modal without crashing 1`] = `
<div
  class="modal"
  style="display: none;"
>
  <div
    class="modal-content"
  >
    <span
      class="close"
    >
      Ã—
    </span>
    <div>
      <div
        class="modal-flex"
      />
    </div>
  </div>
  <div
    class="modal-background"
  />
</div>
`;
